libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -719,55 +719,70 @@ method publish*(
 
   var peers: HashSet[PubSubPeer]
 
-  # add always direct peers
-  peers.incl(g.subscribedDirectPeers.getOrDefault(topic))
-
-  if topic in g.topics: # if we're subscribed use the mesh
-    peers.incl(g.mesh.getOrDefault(topic))
-
-  if g.parameters.floodPublish:
-    # With flood publishing enabled, the mesh is used when propagating messages from other peers,
-    # but a peer's own messages will always be published to all known peers in the topic, limited
-    # to the amount of peers we can send it to in one heartbeat
-
-    let maxPeersToFlood =
-      if g.parameters.bandwidthEstimatebps > 0:
-        let
-          bandwidth = (g.parameters.bandwidthEstimatebps) div 8 div 1000
-            # Divisions are to convert it to Bytes per ms TODO replace with bandwidth estimate
-          msToTransmit = max(data.len div bandwidth, 1)
-        max(
-          g.parameters.heartbeatInterval.milliseconds div msToTransmit,
-          g.parameters.dLow,
+  if useCustomConn:
+    if g.customConnCallbacks.isSome:
+      peers.incl(
```

Comment:
Might make sense to add a trace log here to know we're using a customConnCallback (similar to how it's done on line 7333)

Reply:
Added a `trace` log in `makePeersForPublishUsingCustomConn` function to indicate when a custom peer selection callback is being used.